### DISCUSSION TITLE
Open parquet/arrow/csv files directly from the command line

### DISCUSSION THREAD

**Discussion by perkfly** [tconbeer/harlequin#314] [Ideas (Feature Requests)]
harlequin should be able to open parquet/arrow files directly, instead of importing them after the app is opened.

```
harlequin path/to/*.parquet
```

**Comment by tconbeer** [comment]
What does it mean to "open" them? Would you want it to "select *"?

**Comment by perkfly** [comment]
Yes, automatically select them into a table.

**Reply by tconbeer** [reply]
I think we'll do this as an option, instead, like:

```bash
harlequin -f my_file.parquet
```

This could maybe also let you pipe from stdin by specifying `-`, like the duckdb cli:

```bash
curl https://api.github.com/repos/tconbeer/harlequin | harlequin -f -
```

**Comment by jspaezp** [comment]
Hi there!

@tconbeer I am also interested in this feature and I think the asiest way to implement it is to spin up an in-memory database using the duckdb adapter. I am still becoming familiar with this project but I would love to start a PR if there is interest in this feature (and you could help me get it past the finish line). LMK what you think!

-Sebastian

RN I have a pretty dirty python script that does something similar (cli to use sql on parquet files) and this is the meat of the implementation

```
def main(args):
    con = setup_duckdb(args)
    res = con.sql(f"CREATE VIEW mytable AS SELECT * FROM read_parquet('{args.file}')") # This is the only line that actually matters :P
    res: duckdb.DuckDBPyRelation = con.sql(args.sql)

    kwargs = DuckDBShowConfig.from_args(args).to_kwargs()
    res.show(**kwargs)

    con.close()

```

**Reply by jspaezp** [reply]
https://github.com/jspaezp/harlequin/pull/1 <- Made a proof of concept over here.
I am not the biggest fan of how nested the `DuckDbAdapter.connect` becomes ... and there is
some design decisions to be made for ergonomics.

1. What name to assign to each table?
2. Do we want to combine tables? If I have a directory full of parquet files, do we want to to:
    a.  catenate them as a single table?
    b.  keep them as a table per file?
    c.  Have an argument to select the behavior?
3. For the more ... esoteric uses ... if I pass a duckdb database AND parquet files, do we want to add them as views? (definitely out of the scope I had planned).

LMK what you think!
